core/domain/stats_services.py

```diff
@@ -16,251 +16,137 @@
-                    if state_name in state_hit_counts else 0),
-            } for state_name in exploration.states
-        },
-    }
+    # Handling state additions and deletions.
+    for change_dict in change_list:
+        if change_dict['cmd'] == exp_domain.CMD_ADD_STATE:
+            exploration_stats.state_stats_mapping[change_dict['state_name']] = (
+                stats_domain.StateStats.create_default())
 
+        if change_dict['cmd'] == exp_domain.CMD_DELETE_STATE:
+            exploration_stats.state_stats_mapping.pop(change_dict['state_name'])
 
-def record_answer(
-        exploration_id, exploration_version, state_name, interaction_id,
-        submitted_answer):
-    """Record an answer by storing it to the corresponding StateAnswers entity.
+    exploration_stats.exp_version = new_exp_version
 
-    Args:
-        exploration_id: str. The exploration ID.
-        exploration_version: int. The version of the exploration.
-        state_name: str. The name of the state.
-        interaction_id: str. The ID of the interaction.
-        submitted_answer: SubmittedAnswer. The submitted answer.
-    """
-    record_answers(
-        exploration_id, exploration_version, state_name, interaction_id,
-        [submitted_answer])
+    # Create new statistics model.
+    create_stats_model(exploration_stats)
 
 
-def record_answers(
-        exploration_id, exploration_version, state_name, interaction_id,
-        submitted_answer_list):
-    """Optimally record a group of answers using an already loaded exploration..
-    The submitted_answer_list is a list of SubmittedAnswer domain objects.
+def get_exploration_stats_by_id(exp_id, exp_version):
+    """Retrieves the ExplorationStats domain object.
 
     Args:
-        exploration_id: str. The exploration ID.
-        exploration_version: int. The version of the exploration.
-        state_name: str. The name of the state.
-        interaction_id: str. The ID of the interaction.
-        submitted_answer_list: list(SubmittedAnswer). The list of answers to be
-            recorded.
-    """
-    state_answers = stats_domain.StateAnswers(
-        exploration_id, exploration_version, state_name, interaction_id,
-        submitted_answer_list)
-    for submitted_answer in submitted_answer_list:
-        submitted_answer.validate()
+        exp_id: str. ID of the exploration.
+        exp_version: int. Version of the exploration.
 
-    stats_models.StateAnswersModel.insert_submitted_answers(
-        state_answers.exploration_id, state_answers.exploration_version,
-        state_answers.state_name, state_answers.interaction_id,
-        state_answers.get_submitted_answer_dict_list())
+    Returns:
+        ExplorationStats. The domain object for exploration statistics.
+
+    Raises:
+        Exception: Entity for class ExplorationStatsModel with id not found.
+    """
+    model_id = exp_id + '.' + str(exp_version)
+    exploration_stats_model = stats_models.ExplorationStatsModel.get(
```

Comment:
Should be get(exp_id, exp_version); leave model id construction to the model class since that's an internal thing

Reply:
Done.